packages/shared/src/components/LegoReferralBadge.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { ReactElement } from 'react';
+import classNames from 'classnames';
+import { ReferralOriginKey } from '../graphql/users';
+import { useReferralCampaign } from '../hooks';
+import { Image } from './image/Image';
+import { FlexRow } from './utilities';
+import { useActions } from '../hooks/useActions';
+import { ActionType } from '../graphql/actions';
+
+type LegoReferralBadgeProps = {
+  className?: string;
+  referralTargetCount: number;
+};
+
+const LegoReferralBadge = ({
+  className,
+  referralTargetCount,
+}: LegoReferralBadgeProps): ReactElement => {
+  const { checkHasCompleted } = useActions();
+  const { referredUsersCount, isLoading } = useReferralCampaign({
+    referralOrigin: ReferralOriginKey.LegoMay2023,
+  });
+  const referralCurrentCount =
+    referredUsersCount > referralTargetCount
+      ? referralTargetCount
+      : referredUsersCount;
+
+  if (checkHasCompleted(ActionType.LegoMay2023Hide)) {
+    return null;
+  }
+
+  if (isLoading) {
+    return null;
+  }
+
+  return (
+    <button
+      type="button"
+      className={classNames(
+        className,
+        'rounded-12 p-[3px] bg-gradient-to-br from-[#6DE8BE] via-[#DD3DFC] to-[#F9DD53]',
```

Comment:
The colors here are on purpose different then our color scheme, [checked with design](https://dailydotdev.slack.com/archives/C055LHZ8F9V/p1685599669069279?thread_ts=1685547750.843239&cid=C055LHZ8F9V).